### ISSUE TITLE
file handle  is open after uploading in chunks

### ISSUE THREAD

**Issue by SexyCatGirls** [googleapis/google-api-python-client#596]
Im using python 2.7 with the google api python class to upload multiple videos.
after uploading it i want to delete the file which fails as the handler is still open.
my code is following:

```python
the_file_to_upload = episode + ".mp4"
metadata = {'name': episode + ".mp4", 'parents': [current_folder_id]}
# Note the chunksize restrictions given in
# https://developers.google.com/api-client-library/python/guide/media_upload
media = MediaFileUpload(
    the_file_to_upload,
    chunksize=1024 * 1024 * 8,
    # Not sure whether or not this mimetypes is necessary:
    #mimetype=mime.from_file(the_file_to_upload),
    resumable=True)
request = service.files().create(body=metadata, media_body=media)
response = None
while response is None:
    status, response = request.next_chunk()
    if status:
        print("Uploaded %d%%." % int(status.progress() * 100))
print("Upload of {} is complete.".format(the_file_to_upload))
os.remove(the_file_to_upload)
```

os.remove will fail in this case cause it is still in use from MediaFileUpload.

**Comment by tseaver** [comment]
@SexyCatGirls you can get access to the underlying file object using the `stream()` method (inherited from `MediaIoBaseUpload`).   E.g.:

```python
media.stream().close()
os.remove(the_file_to_upload)
```

**Comment by tseaver** [comment]
One could argue that `MediaFileUpload` should expose a `close` method, since it is the party responsible for opening the file.

**Comment by tristanthomas** [comment]
> @SexyCatGirls you can get access to the underlying file object using the `stream()` method (inherited from `MediaIoBaseUpload`). E.g.:
> 
> ```python
> media.stream().close()
> os.remove(the_file_to_upload)
> ```

Thank you! This fixed my issue: After uploading a file using MediaFileUpload, it kept the file descriptor open. When I try to move, rename, or change the original file afterwards, I encounter the error "[Error 32] The process cannot access the file because it is being used by another process". Note: I only ran into this issue on Windows. I did not have this issue under macOS and Linux.
